app/services/sync_ndbn_members.rb

```diff
@@ -1,69 +1,45 @@
 class SyncNDBNMembers
-  include HTTParty
+  include ServiceObjectErrorsMixin
 
-  NDBN_MEMBERS_PAGE = "https://ndbn.memberclicks.net/member-ids".freeze
+  def initialize(member_file)
+    @member_file = member_file
+  end
+  attr_reader :member_file
 
-  class << self
-    def sync
-      Rails.logger.info("Syncing NDBNMember started.")
+  def call
+    member_entries = parse_csv
 
-      members_page = fetch_members_page
-      member_entries = extract_member_entries(members_page)
+    return if errors.any?
 
-      member_entries.each do |member|
-        ndbn_member = NDBNMember.find_or_initialize_by(ndbn_member_id: member[:ndbn_member_id])
-        ndbn_member.account_name = member[:account_name]
+    member_entries.each do |member_id, account_name|
+      ndbn_member = NDBNMember.find_or_initialize_by(ndbn_member_id: member_id.to_i)
+      ndbn_member.account_name = account_name
 
-        # Skip if nothing has changed!
-        if ndbn_member.persisted? && !ndbn_member.changed?
-          next
-        elsif ndbn_member.persisted? && ndbn_member.changed?
-          Rails.logger.info("#{ndbn_member.id} changed their account name from #{ndbn_member.account_name_was} to #{ndbn_member.account_name}.")
-        elsif ndbn_member.new_record?
-          Rails.logger.info("New record found! #{ndbn_member.id} - #{ndbn_member.account_name} has been added!")
-        end
+      # Skip if nothing has changed!
+      if ndbn_member.persisted? && !ndbn_member.changed?
```

Comment:
Not sure you need this - Rails is generally smart enough not to try to update a record if nothing has changed.